src/main/java/note/NoteList.java

```diff
@@ -118,21 +127,37 @@ public void openNoteProcess() {
         boolean checkExistance = ui.printOpenNoteMessage(this);
         if (checkExistance) {
             String userInput = ui.readUserInput();
-            if (userInput.contains("search")) {
+            if (userInput.contains("1")) {
                 ui.printNoteSearchInstructions();
-                userInput = ui.readUserInput();
-                selectSearchMethod(userInput);
-            } else {
-                openNoteDirectly();
+
+                while (!userInput.equals("")) {
+                    userInput = ui.readUserInput();
+                    try {
+                        selectSearchMethod(userInput);
+                    } catch (InvalidNoteSearchException e) {
+                        ui.printNoteSearchError(INVALID_SEARCH_OPTION);
+                    }
+                }
+            } else if (userInput.contains("2")) {
+                while (!userInput.equals("")) {
+                    try {
+                        openNoteDirectly();
+                        break;
+                    } catch (IndexOutOfBoundsException e) {
+                        ui.printNoteMissingError(notes.size());
+                    }
+                }
             }
         }
     }
```

Comment:
I think this is too deeply nested? Maybe can refactor to extract out methods

Reply:
I do not think this one should be refactored. Because the while loop is used for allowing user to re-input.

Reply:
Ok no problem, it's up to you :)